src/dialogs/RoomSettings.cpp

```diff
@@ -59,13 +59,59 @@ RoomSettings::RoomSettings(const QString &room_id, QWidget *parent)
         notifOptionLayout_->addWidget(notifLabel);
         notifOptionLayout_->addWidget(notifCombo, 0, Qt::AlignBottom | Qt::AlignRight);
 
+        auto accessOptionLayout = new QHBoxLayout();
+        accessOptionLayout->setMargin(5);
+        auto accessLabel = new QLabel(tr("Room access"), this);
+        accessCombo = new QComboBox(this);
+        accessCombo->addItem(tr("Anyone and guests"));
+        accessCombo->addItem(tr("Anyone"));
+        accessCombo->addItem(tr("Inviteds"));
+        accessCombo->setDisabled(true);
+        accessLabel->setStyleSheet("font-size: 15px;");
+
+        if(info_.join_rule == JoinRule::Public)
+                if(info_.guest_access)
+                        accessCombo->setCurrentIndex(0);
+                else
+                        accessCombo->setCurrentIndex(1);
+        else
+                accessCombo->setCurrentIndex(2);
+
+        accessOptionLayout->addWidget(accessLabel);
+        accessOptionLayout->addWidget(accessCombo);
+
         layout->addWidget(new TopSection(info_, avatarImg_, this));
         layout->addLayout(notifOptionLayout_);
         layout->addLayout(notifOptionLayout_);
+        layout->addLayout(accessOptionLayout);
         layout->addLayout(btnLayout);
 
         connect(cancelBtn_, &FlatButton::clicked, this, &RoomSettings::closing);
-        connect(saveBtn_, &FlatButton::clicked, this, [this]() { emit closing(); });
+        connect(saveBtn_, &FlatButton::clicked, this, &RoomSettings::save_and_close);
+}
+
+void
+RoomSettings::save_and_close() {
+        // TODO: Save access changes to the room
```

Comment:
To save the changes you'll have to send a state event using the http API. 

https://matrix.org/speculator/spec/HEAD/client_server/unstable.html#get-matrix-client-r0-rooms-roomid-state-eventtype-statekey